rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -445,8 +445,7 @@ def parse_value_string(type_, value_string):
     if type_.is_primitive_type() and not type_.is_array:
         return parse_primitive_value_string(type_, value_string)
 
-    # TODO(mikaelarguedas) change this condition once string escape function is implemented
-    if type_.is_primitive_type() and type_.is_array and type_.type != 'string':
+    if type_.is_primitive_type() and type_.is_array:
```

Comment:
I agree with @wjwwood that it would be simpler and clearer if we require users to quote and escape the interior quotes. Not only to reduce parser complexity but also  improve human readability.

I agree that it all comes down to how (what format) we expect people to pass parameters (both CLI and IDL/config files). Do we have a document or discussion somewhere so that we could iterate on and refer to before making a decision on the scope of this parser?